### PR TITLE
evaluator: move the evaluate of ast function trim to buildin

### PR DESCRIPTION
**Pull request by zxylvlp**
Move the evaluate of ast function trim to builtin.

### CODE REVIEW

**Comment by zxylvlp** [comment]
@shenli @coocood  PTAL

**Comment by coocood** [comment]
LGTM

File: evaluator/builtin/string.go [hunk 1]
```diff
@@ -370,3 +371,79 @@ func builtinLocate(args []interface{}, _ context.Context) (interface{}, error) {
...
+		if fs != nil {
+			remstr, err = types.ToString(fs)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+	}
+	// Do trim
+	var result string
+	var drct ast.TrimDirectionType
```
Comment (shenli) [inline evaluator/builtin/string.go]: dir is the abbr. for direction.

**Comment by shenli** [comment]
LGTM

File: evaluator/builtin/string.go [hunk 2]
```diff
@@ -370,3 +371,79 @@ func builtinLocate(args []interface{}, _ context.Context) (interface{}, error) {
...
+	if len(args) > 1 {
+		fs = args[1]
+		if fs != nil {
+			remstr, err = types.ToString(fs)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+	}
+	// Do trim
```
Comment (zimulala) [inline evaluator/builtin/string.go]: Do -> do ?

File: evaluator/builtin/string.go [hunk 3]
```diff
@@ -370,3 +371,79 @@ func builtinLocate(args []interface{}, _ context.Context) (interface{}, error) {
 	}
 	return int64(i) + pos + 1, nil
 }
+
+const spaceChars = "\n\t\r "
+
+// See: https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_trim
```
Comment (zimulala) [inline evaluator/builtin/string.go]: add . at the end of a line

**Pull request merged**